File: runtime/src/de/fatox/meta/api/entity/EntityManager.java
package de.fatox.meta.api.entity;

public interface EntityManager<ENTITY extends Entity> {

    Iterable<ENTITY> getEntities();

    Iterable<ENTITY> getStaticEntities();

    void addEntity(ENTITY entity);

    void addStaticEntity(ENTITY entity);

    void removeEntity(ENTITY entity);

    void clear();
}

File: runtime/src/main/kotlin/de/fatox/meta/api/entity/EntityManager.kt
package de.fatox.meta.api.entity

interface EntityManager<ENTITY : Entity<*>> {

	val entities: Iterable<ENTITY>

	val staticEntities: Iterable<ENTITY>

	fun addEntity(entity: ENTITY)

	fun addStaticEntity(entity: ENTITY)

	fun removeEntity(entity: ENTITY)

	fun clear()
}
